### ISSUE TITLE
Search does not work on deeply nested array of objects

### ISSUE THREAD

**Issue by anirudh1713** [typesense/typesense#1403]
## TL;DR
the search is effective at the first level of nesting but does not work on the second level.

## Description

Search does not work when specific fields of nested objects are indexed -- Both `items` and `items.nested_items` are array of objects.
```json
{
  "facet": false,
  "index": true,
  "infix": false,
  "locale": "",
  "name": "items.name",
  "optional": true,
  "sort": false,
  "type": "string[]"
},
{
  "facet": false,
  "index": true,
  "infix": false,
  "locale": "",
  "name": "items.nested_items.name",
  "optional": true,
  "sort": false,
  "type": "string[]"
}
```

however, it **works** when the entire object is indexed -
```json
{
  "name": "items",
  "type": "object[]"
},
{
  "name": "items.nested_items",
  "type": "object[]"
}
```

## Steps to reproduce
```sh
export TYPESENSE_API_KEY=xyz

curl "http://localhost:8108/debug" \
       -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}"

curl --location "http://localhost:8108/collections" \
--header "Content-Type: application/json" \
--header "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
--data '{
    "enable_nested_fields": true,
    "fields": [
        {
            "facet": false,
            "index": true,
            "infix": false,
            "locale": "",
            "name": "items.name",
            "optional": true,
            "sort": false,
            "type": "string[]"
        },
        {
            "facet": false,
            "index": true,
            "infix": false,
            "locale": "",
            "name": "items.description",
            "optional": true,
            "sort": false,
            "type": "string[]"
        },
        {
            "facet": false,
            "index": true,
            "infix": false,
            "locale": "",
            "name": "items.nested_items.name",
            "optional": true,
            "sort": false,
            "type": "string[]"
        }
    ],
    "name": "test"
}'

curl --location 'http://localhost:8108/collections/test/documents?action=upsert' \
--header "Content-Type: application/json" \
--header "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
--data '{
    "id": "7aae6a64-1dd0-4fd5-a6aa-f2807e804222",
    "items": [
        {
            "description": "random description.",
            "id": "d568961d-d555-4d22-b905-7543016ca695",
            "name": "pirates",
            "nested_items": [
                {
                    "isAvailable": true
                },
                {
                    "isAvailable": true
                },
                {
                    "isAvailable": true,
                    "name": "one piece"
                },
                {
                    "description": "nested description here",
                    "isAvailable": true,
                    "name": "naruto"
                },
                {
                    "description": "description again", 
                    "isAvailable": true,
                    "name": "dragon ball"
                }
            ]
        },
        {
            "description": "random description.",
            "id": "e568961d-d555-4d22-b905-7543016ca699",
            "name": "super nova",
            "nested_items": [
                {
                    "isAvailable": true
                },
                {
                    "isAvailable": true
                },
                {
                    "isAvailable": true,
                    "name": "star wars"
                },
                {
                    "description": "nested description here",
                    "isAvailable": true,
                    "name": "pokemon"
                },
                {
                    "description": "description again",
                    "isAvailable": true,
                    "name": "ben 10"
                }
            ]
        }
    ]
}'

curl --location 'http://localhost:8108/collections/test/documents/search?q=naruto&query_by=items.name%2Citems.nested_items.name' \
--header "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}"
```

## Expected Behavior
The last `search` call should return the first document as one of the `items.nested_items.name` has value `naruto`.

However, if you search for `pirates` you'll get the first document as expected.

## Actual Behavior
It does not return any hits.

## Metadata

**Typesense Version**: `0.25.1`

**OS**: `macOS 14.1 (23B2073)`

**Comment by kishorenc** [comment]
@anirudh1713 

We've a fix in this build: `0.25.2.rc11`.

Can you please check and let us know?

**Comment by anirudh1713** [comment]
@kishorenc Tested yesterday and its working fine!

Thanks a lot for this 🙌
